x/blob/types/blob_tx.go

```diff
@@ -0,0 +1,118 @@
+package types
+
+import (
+	"bytes"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/nmt/namespace"
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+// ProcessedBlobTx caches the unmarshalled result of the BlobTx
+type ProcessedBlobTx struct {
+	Blobs []tmproto.Blob
+	Tx    []byte
+}
+
+// NewBlob creates a new coretypes.Blob from the provided data after performing
+// basic stateless checks over it.
+func NewBlob(ns namespace.ID, blob []byte) (*tmproto.Blob, error) {
+	err := ValidateBlobNamespaceID(ns)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(blob) == 0 {
+		return nil, ErrZeroBlobSize
+	}
+
+	return &tmproto.Blob{
+		NamespaceId:  ns,
+		Data:         blob,
+		ShareVersion: uint32(appconsts.ShareVersionZero),
+	}, nil
+}
+
+// ProcessBlobTx performs stateless checks on the BlobTx to ensure that the
+// blobs attached to the transaction are valid. During this process, it
+// separates the blobs from the MsgPayForBlob, which are returned in the
+// ProcessedBlobTx.
+//
+// NOTE: ProcessBlobTx does not call the ValidateBasic method on either the
+// transaction or messages in that transaction.
+func ProcessBlobTx(txcfg client.TxEncodingConfig, bTx tmproto.BlobTx) (ProcessedBlobTx, error) {
+	sdkTx, err := txcfg.TxDecoder()(bTx.Tx)
+	if err != nil {
+		return ProcessedBlobTx{}, err
+	}
+
+	msgs := sdkTx.GetMsgs()
+	pfbs := make([]*MsgPayForBlob, 0)
+	for _, msg := range msgs {
+		if sdk.MsgTypeURL(msg) == URLMsgPayForBlob {
+			pfb, ok := msg.(*MsgPayForBlob)
+			if !ok {
+				continue
```

Comment:
Shouldn't this error? If this is not true isn't this an incorrectly constructed `BlobTx`?

Reply:
yes, great point. we can and should reject all transaction that one or more invalid PFBs. 

fixed here 4d712c9e21ccaea81531e8d52cd34305b62fefd5